academy/ibc-dev/ibcapp_steps.md

```diff
@@ -0,0 +1,425 @@
+# Make a module IBC-enabled
+
+In this section, you'll build a conceptual SDK blockchain with one module. The first time, a regular module and the second time an IBC module. This will introduce us to what makes a module IBC-enabled.
+
+## Scaffold a Leaderboard chain
+
+By now, you are familiar with scaffolding a chain with Ignite CLI (if not, check out the [Create Your Own Chain](insert_link) section).
+
+Let's scaffold a `leaderboard` chain:
+
+```bash
+ignite scaffold chain github.com/cosmonaut/leaderboard
+```
+
+This creates a chain with `x/leaderboard` a regular SDK module.
+
+Next, scaffold another chain (for example in another git branch) but decide to add the `--no-module` flag:
+
+```bash
+ignite scaffold chain github.com/cosmonaut/leaderboard --no-module
+```
+
+Now we add the `x/leaderboard` module as an IBC module with the `--ibc` flag:
+
+```bash
+ignite scaffold module leaderboard --ibc
```

Comment:
A thought I had. In "my own chain", should the exercise be in order:

* Create the checkers chain up to "wager denomination". At this point, we consider it "complete".
* Add the CosmJS elements for the "complete" chain.
* Create a `module leaderboard --ibc` in the checkers project.
* Describe the migration process with a new module.

Reply:
At the moment, we do have a working version of the checkers game with a leaderboard and IBC enabled in the checkers module. You can send the player score from one checkers chain to another and get your score included into the leaderboard. 
This is going to be two separate chains, a leaderboard chain maintaining the leaderboard and a checkers chain being able to send the score to the leaderboard chain. The idea is to allow other game chains(not just checkers) to send player infos to the "global" leaderboard chain. 
But instead we could also provide a leaderboard module which could be included by the game chains. The issue I see is that we will have different leaderboards maintained across different chains.

Reply:
The IBC extension of the checkers blockchain would be a standalone tutorial, so in that sense it could make sense to keep the original content as is. Then the IBC tutorial could present itself as a refactor starting from the commit that completed the checkers chain up until the wager denomination

Then instead of adding leaderboard as a data structure within the checkers module we refactor to make it a separate chain as @murataniloener pointed out.